### PR TITLE
Reactify Task List

### LINKED ISSUE #1384: Search field on task query page is confusing
Users confuse both search fields. Eq. instead of using the query field they enter an ID into the searchbar to find a task (won't work because that searchbar only searchs through the results...)

 Reported by: Heiko Wissler (hwissler@brain.mpg.de) on master

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/1384/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by hotzenklotz**
I reactified the task list and got rid of the weird query interface and replaced it with a regular form.

### Mailable description of changes:
- [Admin.Task List ] Modernized table and added more filtering / sorting options
- [Admin.Task List ] Replaced complicated "task query" search interface with form
- [Admin.Task List ] Enabled searching for active tasks for specific users

### Steps to test:
- Go to Admin --> tasks
- Search for tasks and double check if the response is correct (e.g. double check with your DB)

### Issues:
- fixes #1384 

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/admin/views/task/task_list_view.js [hunk 1]
```diff
@@ -1,131 +1,256 @@
...
+                    {task.status.completed > 0 ? (
+                      <a
+                        href={`/annotations/CompoundTask/${task.id}`}
+                        title="View all Finished Tracings"
+                      >
+                        <Icon type="eye-o" />View
+                      </a>
+                    ) : null}
+                    <br />
+                    <a href={`/tasks/${task.id}/edit`} title="Edit Project">
```
Comment (daniel-wer) [inline app/assets/javascripts/admin/views/task/task_list_view.js]: The copy and paste is strong in this one :D

File: app/assets/javascripts/admin/views/task/task_search_form.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            <FormItem {...formItemLayout} label="Task Id">
+              {getFieldDecorator("taskId")(<Input placeholder="One or More Task IDs" />)}
+            </FormItem>
+          </Col>
+          <Col span={12}>
+            <FormItem {...formItemLayout} label="Task Type">
+              {getFieldDecorator("taskTypeId")(
+                <Select
+                  showSearch
+                  placeholder="Select a Task Type"
```
Comment (daniel-wer) [inline app/assets/javascripts/admin/views/task/task_search_form.js]: Please add `allowClear` here :)

Comment (daniel-wer) [reply app/assets/javascripts/admin/views/task/task_search_form.js]: And for the other Select Components.

**Comment by hotzenklotz** [comment]
@daniel-wer I applied the frontend changes :-)

**Pull request merged**